From 8c666347ce75de72f2d5108b70752e48af800d72 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Fri, 05 Mar 2021 12:41:22 +0200
Subject: [PATCH] [SW-20154] check for srns and ssc_en masks per platform

SRNS and SSC_EN bits are specified by using different
bits of the strap register, for each platform. Hence
move this to platform specific code.

Change-Id: I1f20d0252a02188398f90f3e9db0b9d603c48ccf
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/pcie.c b/common/pcie.c
index 5176de9..50c8f4d 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -33,19 +33,18 @@
 static void pcie_wa_snps_case_01038891(uint32_t boot_strap)
 {
 	uint8_t separate_reference = 0;
-	uint8_t ssc_en = 0;
+	uint8_t ssc_en = 0, srns = 0;
 
 	separate_reference = get_sris_mode_from_strap();
 
 	/* SRNS mode enable */
-	if ((boot_strap &
-	     PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_BTL_EN_MASK) == 0)
+	srns = get_srns_mode_from_strap(boot_strap);
+
+	if (!srns)
 		separate_reference = 1;
 
 	/* spread spectum clock enable */
-	if (boot_strap &
-	    PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_BTL_ROM_EN_MASK)
-		ssc_en = 1;
+	ssc_en = get_ssc_en_from_strap(boot_strap);
 
 	if ((arcboot_rom_global.config_fuse &
 	     EFUSE_SYSCFG_WA_SNPS_CASE_01038891) == 0) {
diff --git a/common/pcie.h b/common/pcie.h
index 42ca652..1eb8881 100644
--- a/common/pcie.h
+++ b/common/pcie.h
@@ -26,5 +26,7 @@
 void pcie_phy_set_mpll_a_b_mul_bw(uint8_t separate_reference);
 void pcie_phy_set_ext_tx_rx_ctrl(void);
 void pcie_phy_set_protocol0_ext_tx_rx(void);
+uint8_t get_ssc_en_from_strap(uint32_t boot_strap);
+uint8_t get_srns_mode_from_strap(uint32_t boot_strap);
 
 #endif /* _PCIE_H_ */
diff --git a/gaudi2_arc_gnu/common.c b/gaudi2_arc_gnu/common.c
index 32fff06..21e7725 100644
--- a/gaudi2_arc_gnu/common.c
+++ b/gaudi2_arc_gnu/common.c
@@ -432,3 +432,26 @@
 
 void pcie_phy_set_ext_tx_rx_ctrl(void) {}
 void pcie_phy_set_protocol0_ext_tx_rx(void) {}
+
+#define DUMP_SEL_SRNS_MASK	0x1
+#define DUMP_SEL_SSC_MASK	0x2
+
+uint8_t get_srns_mode_from_strap(uint32_t boot_strap)
+{
+	uint32_t dump_sel = (boot_strap &
+		PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_DUMP_SEL_MASK) >>
+		PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_DUMP_SEL_SHIFT;
+
+	/* dumpsel[0] is SRNS */
+	return (dump_sel & DUMP_SEL_SRNS_MASK);
+}
+
+uint8_t get_ssc_en_from_strap(uint32_t boot_strap)
+{
+	uint32_t dump_sel = (boot_strap &
+		PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_DUMP_SEL_MASK) >>
+		PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_DUMP_SEL_SHIFT;
+
+	/* dumpsel[1] is SSC EN */
+	return !!(dump_sel & DUMP_SEL_SSC_MASK);
+}
diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index fc55336..b6ea28d 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -157,3 +157,13 @@
 	regmap_writel(PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4_CONF,
 			mmPCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4);
 }
+uint8_t get_srns_mode_from_strap(uint32_t boot_strap)
+{
+	return !!(boot_strap & PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_BTL_EN_MASK);
+}
+
+uint8_t get_ssc_en_from_strap(uint32_t boot_strap)
+{
+	return (!!(boot_strap &
+			PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_BTL_ROM_EN_MASK));
+}
